lib/puppet/pal/pal_impl.rb

```diff
@@ -58,8 +58,8 @@ def self.with_script_compiler(
       configured_by_env: false,
       manifest_file:     nil,
       code_string:       nil,
-      facts:             nil,
-      variables:         nil,
+      facts:             {},
+      variables:         {},
```

Comment:
Is this an API behavior change (such as if older bolt calls into newer puppet)? It looks like `nil` and `Hash#empty?` are treated the same below, but just wanted to confirm.

Reply:
I think since this is just used by Bolt and PuppetServer (at least I very much hope!) we're not too worried about the API change. I double check in Puppetserver that they don't rely on these being nil, and they don't, and also gave the team a heads up that we were changing this API. Definitely a good concern to raise though! If you know of anywhere else these compilers are used I would love to know!